system/lib/pthread/library_pthread.c

```diff
@@ -135,6 +135,30 @@ static void em_queued_call_free(em_queued_call* call) {
   free(call);
 }
 
+static void init_em_queued_call_args(em_queued_call* q,
+                                     EM_FUNC_SIGNATURE sig,
+                                     va_list args) {
+  EM_FUNC_SIGNATURE argumentsType = sig & EM_FUNC_SIG_ARGUMENTS_TYPE_MASK;
```

Comment:
I think you can do `q->functionEnum = sig;` here too..   in fact maybe just pass func_ptr too and call this `init_em_queued_call`

Reply:
In an upcoming PR I add a function similar to what you suggest that allocates and initializes the full `em_queued_call`. The reason to bundle the initialization with allocation is that it is much easier to manually initialize a stack-allocated `em_queued_call`, so an initialization function (except for the `args`) is not as useful.